src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java

```diff
@@ -19,6 +21,8 @@
 public class MqttMessage implements Message {
     String topic;
     byte[] payload;
+    @NonFinal
```

Comment:
everything is supposed to be final. To change a value use `toBuilder` and then `withRetain`